awstesting/integration/performance/s3DownloadManager/main.go

```diff
@@ -216,3 +236,99 @@ func newDownloader(clientConfig ClientConfig, sdkConfig SDKConfig, options ...re
 
 	return downloader
 }
+
+func getObjectSize(bucket, key string) (int64, error) {
+	sess := session.Must(session.NewSession())
+	svc := s3.New(sess)
+	resp, err := svc.HeadObject(&s3.HeadObjectInput{
+		Bucket: &bucket,
+		Key:    &key,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	return *resp.ContentLength, nil
+}
+
+type Profiler struct {
+	outputDir string
+
+	enableCPU   bool
+	enableTrace bool
+
+	cpuFile   *os.File
+	traceFile *os.File
+}
+
+func (p *Profiler) SetupFlags(prefix string, flagSet *flag.FlagSet) {
+	prefix += "profiler."
+
+	flagSet.StringVar(&p.outputDir, prefix+"output-dir", os.TempDir(), "output directory to write profiling data")
+	flagSet.BoolVar(&p.enableCPU, prefix+"cpu", false, "enable CPU profiling")
+	flagSet.BoolVar(&p.enableTrace, prefix+"trace", false, "enable tracing")
+}
+
+func (p *Profiler) Start() {
+	var err error
+
+	uuid := integration.UniqueID()
+	if p.enableCPU {
+		p.cpuFile, err = p.createFile(uuid, "cpu")
+		if err != nil {
+			panic(fmt.Sprintf("failed to create cpu profile file: %v", err))
+		}
+		err = pprof.StartCPUProfile(p.cpuFile)
+		if err != nil {
+			panic(fmt.Sprintf("failed to start cpu profile: %v", err))
+		}
+	}
+	if p.enableTrace {
+		p.traceFile, err = p.createFile(uuid, "trace")
+		if err != nil {
+			panic(fmt.Sprintf("failed to create trace file: %v", err))
+		}
+		err = trace.Start(p.traceFile)
+		if err != nil {
+			panic(fmt.Sprintf("failed to tracing: %v", err))
+		}
+	}
+}
+
+func (p *Profiler) Stop() {
+	logAndCloseFile := func(profile string, file *os.File) {
```

Comment:
does it make sense for this to be an unexported method or static func on `Profiler` instead of a dynamic closure?